clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -275,8 +275,10 @@ private void handleProduceResponse(ClientResponse response, Map<TopicPartition,
                 for (Map.Entry<TopicPartition, ProduceResponse.PartitionResponse> entry : produceResponse.responses().entrySet()) {
                     TopicPartition tp = entry.getKey();
                     ProduceResponse.PartitionResponse partResp = entry.getValue();
-                    ProducerBatch batch = batches.get(tp);
-                    completeBatch(batch, partResp, correlationId, now);
+                    if (tp != null && partResp != null) {
+                        ProducerBatch batch = batches.get(tp);
+                        completeBatch(batch, partResp, correlationId, now);
+                    }
```

Comment:
That's right. But as far as I can see, it doesn't provide enough detail to pinpoint that this would fix it. What I am missing?